app/controllers/notification_controller.rb

```diff
@@ -0,0 +1,31 @@
+class NotificationController < ApplicationController
+  before_action :fetch_conference_room
+  before_action :validate_request
+
+  CHANNEL_ID_HEADER = 'X-Goog-Channel-ID'.freeze
+
+  def receive
+    NotifyClientsJob.perform_later params[:conference_room_id]
+    head :ok
+  end
+
+  private
+
+  def validate_request
+    head :bad_request unless valid_request?
+  end
+
+  def valid_request?
+    @conference_room.channel && @conference_room.channel.channel_id == channel_id
```

Comment:
You could add: `delegate :channel_id, to: :channel, allow_null: true` to ConferenceRoom model.

Then you don't have to check for `@conference_room.channel` existence and you don't violate Law of Demeter in `@conference_room.channel.channel_id`

 See (http://apidock.com/rails/Module/delegate)